model/goniometer.py

```diff
@@ -223,25 +231,35 @@ def multi_axis_goniometer_from_phil(params, reference=None):
     @staticmethod
     def from_phil(params, reference=None):
         """
-        Convert the phil parameters into a beam model
+        Convert the phil parameters into a goniometer model
 
         """
+        if params.goniometer.axis and params.goniometer.axes:
+            raise ValueError("Only one of axis or axes should be set")
         if reference is not None:
             if isinstance(reference, MultiAxisGoniometer):
+                if params.goniometer.axis:
+                    raise ValueError(
+                        "Cannot set 'axis' with a multi-axis reference goniometer"
+                    )
                 goniometer = GoniometerFactory.multi_axis_goniometer_from_phil(
                     params, reference
                 )
             else:
+                if params.goniometer.axes:
+                    raise ValueError(
+                        "Cannot set 'axes' with a single-axis reference goniometer"
```

Comment:
Do we want to stop people using the `axes` parameter to specify the axis for a single-axis goniometer? Previously this did work (in fact, it was the only way).

Reply:
Hmm... maybe? It is an API change though and is mostly harmless. We should deprecate it first if we decide to remove.